Build/rpm/stepmania5.spec

```
@@ -0,0 +1,36 @@
+Name:       stepmania5
+Version:    0
+Release:    12
+Summary:    A dance and rhythm game
+License:    GPL
+BuildRequires: gcc-c++ cmake libmad-devel libvorbis-devel libbz2-devel libX11-devel libjpeg62-devel libXtst-devel libpulse-devel alsa-devel libva-devel glew-devel libXrandr-devel libXinerama-devel
+Requires:      glibc libasound2 libbz2-1 libdbus-1-3 libFLAC8 libgcc_s1 libgcrypt20 libGLEW2_2 libGLU1 libglvnd libgpg-error0 libjpeg62 liblz4-1 liblzma5 libmad0 libogg0 libpulse0 libsndfile1 libspeex1 libstdc++6 libsystemd0 libva2 libvorbis0 libvorbisenc2 libvorbisfile3 libX11-6 libXau6 libxcb1 libXext6 libXinerama1 libXrandr2 libXrender1 libXtst6
+%description
+StepMania is a free dance and rhythm game. It features 3D graphics, keyboard and "dance pad" support, and an editor for creating your own steps.
+
+%prep
+#--------------------------------------------------------------------------------
+#--- If repo exists locally on ~/git/stepmania it will be checkout from there ---
+#--- Otherwise it will be checked-out from the official repository URL        ---
+#--------------------------------------------------------------------------------
+rm -rf %{_sourcedir}/*
+git clone ~/git/stepmania %{_sourcedir} || git clone https://github.com/stepmania/stepmania.git %{_sourcedir}
+
+%build
+cd %{_builddir}
+cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=%{buildroot}/opt %{_sourcedir}
```

I was trying to find the most resource-reusing (and compiling quickly) build options that would work on Mageia linux, and found the following (enabling all the `WITH_SYSTEM_*`) to work perfectly fine. Feel free to use them if you like (some may require updating BuildRequires and/or Requires, but some are already there, like JPEG/MAD/OGG...)

```suggestion
cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=%{buildroot}/opt %{_sourcedir} \
  -DWITH_SYSTEM_FFMPEG=1 \
  -DWITH_SYSTEM_GLEW=1 \
  -DWITH_SYSTEM_JPEG=1 \
  -DWITH_SYSTEM_JSONCPP=1 \
  -DWITH_SYSTEM_MAD=1 \
  -DWITH_SYSTEM_OGG=1 \
  -DWITH_SYSTEM_PCRE=1 \
  -DWITH_SYSTEM_PNG=1 \
  -DWITH_SYSTEM_TOMCRYPT=1 \
  -DWITH_SYSTEM_TOMMATH=1 \
  -DWITH_SYSTEM_ZLIB=1
```